src/components/Tree/Tree.cy.tsx

```diff
@@ -34,37 +66,231 @@ const TreeComponent = ({ onSelect }: { onSelect?: (id: Nullable<string>) => void
+
+        cy.get(TREE_ITEM_TOGGLE_ID).first().click();
+        cy.get(TREE_ITEM_ID).should('have.length', 8);
+        cy.get(TREE_ITEM_DRAG_HANDLE_ID).first().realMouseDown();
+        cy.get(TREE_ITEM_ID).should('have.length', 3);
+        cy.get(TREE_ITEM_DRAG_HANDLE_ID).first().realMouseUp();
+    });
+
+    it('should trigger onDrop with the keyboard as a controlled component', () => {
+        const onDropStub = cy.stub().as('onDropStub');
+
+        cy.mount(<TreeComponent draggable onDropTree={onDropStub} />);
+
+        cy.get(TREE_ITEM_DRAG_HANDLE_ID)
+            .first()
+            .focus()
+            .realPress('Space')
+            .realPress('ArrowDown')
+            .realPress('ArrowRight')
+            .realPress('Space');
+
+        cy.get('@onDropStub').should('have.been.calledOnce');
+    });
+
+    it('should trigger onDrop on the root item as a controlled component', () => {
+        const onDropStub = cy.stub().as('onDropStub');
+
+        cy.mount(<TreeComponent draggable onDropTreeItem={onDropStub} />);
+
+        cy.get(TREE_ITEM_DRAG_HANDLE_ID).first().realMouseDown().realMouseMove(0, 40).realMouseUp();
+
+        cy.get('@onDropStub').should('have.been.calledOnce');
+    });
+
+    it('should trigger onDrop on a level 1 item as a controlled component', () => {
+        const onDropStub = cy.stub().as('onDropStub');
+
+        cy.mount(<TreeComponent draggable onDropTreeItem={onDropStub} />);
+
+        cy.get(TREE_ITEM_TOGGLE_ID).first().click();
+        cy.get(TREE_ITEM_DRAG_HANDLE_ID).eq(4).realMouseDown().realMouseMove(0, -40).realMouseUp();
+
+        cy.get('@onDropStub').should('have.been.calledOnce');
+    });
+
+    it('should not trigger onDrop when moving to an not allowed inside item', () => {
+        const onDropStub = cy.stub().as('onDropStub');
 
+        cy.mount(<TreeComponent draggable onDropTreeItem={onDropStub} />);
+
+        cy.get(TREE_ITEM_TOGGLE_ID).first().click();
+
+        cy.get(TREE_ITEM_DRAG_HANDLE_ID).eq(6).realMouseDown().realMouseMove(40, -40).realMouseUp();
+        cy.get('@onDropStub').should('not.have.been.called');
+    });
+
+    it('allows adding an item dynamically to the Tree', () => {
+        const newNode: TreeItemMock = {
+            id: '4',
+            label: 'New Added Item',
+        };
+        const rootItemsLength = treeItemsMock.length;
+
+        cy.mount(<TreeComponent />).then(({ rerender }) => {
+            cy.get(TREE_ITEM_ID).should('have.length', rootItemsLength);
+            treeItemsMock.push(newNode);
+            rerender(<TreeComponent />);
+            cy.get(TREE_ITEM_ID).should('have.length', rootItemsLength + 1);
+        });
+    });
+
+    it('allows removing an item dynamically to the Tree', () => {
+        const rootItemsLength = treeItemsMock.length;
+
+        cy.mount(<TreeComponent />).then(({ rerender }) => {
+            cy.get(TREE_ITEM_ID).should('have.length', rootItemsLength);
+            treeItemsMock.pop();
```

Comment:
It would be safer if we reset the treeItemsMock array after push/pop in test cases for keeping the integrity of the mocks for the next cases. WDYT?

Reply:
done